Add API Readiness Checklist content (#392)
camaraproject/ReleaseManagement: documentation/readiness/api-readiness-checklist.md

## Purpose

Before an API repository can be released, codeowners must ensure that certain assets are in place and meet quality expectations appropriate for the declared target API status. This checklist:

- Defines the required release assets and their expected locations
- Specifies which assets are mandatory (M) or optional (O) per API status
- Clarifies the division of responsibility: **codeowners prepare** the assets, **automation validates** what it can, and **release management reviewers** verify the rest

Release readiness is tracked through `release-plan.yaml` configuration, release content preparation as indicated in the Release Issue, and the review checklist in the Release PR.

## Terms

| Term | Source | Values |
|------|--------|--------|
| Release type | `repository.target_release_type` | `pre-release-alpha`, `pre-release-rc`, `public-release`, `maintenance-release` |
| API status | `apis[].target_api_status` | `alpha`, `rc`, `public` |
| Initial public | Derived | API status is `public` and major version is 0 (e.g., v0.5.0) |
| Stable public | Derived | API status is `public` and major version ≥ 1 (e.g., v1.0.0) |

> The Release Issue and Release PR embed the readiness matrix as a convenience copy; this document is the canonical source.

Reviewer: ```suggestion
The Release Issue and Release PR embed the readiness matrix as a convenience copy; this document holds the reference source.
```

| 3 | Commonalities compliance | API definitions follow the Commonalities version declared in `release-plan.yaml` dependencies | In API definitions | Partially (Spectral rules cover structure; design guidelines require manual review) | Spectral covers structure; review design guideline adherence manually |
| 4 | API Documentation | API description in the YAML `info.description` field; additional documentation as needed | In YAML `info` section or `documentation/` | Partial (`info.description` presence checked) | `info.description` is present and meaningful |
| 5 | User Stories | At least one user story per API demonstrating the intended use | `documentation/API_documentation/` | No | At least one user story document per API exists |
| 6 | Test Cases (basic) | Sunny day scenarios and main error cases; at least one `.feature` file per API | `code/Test_definitions/` | File existence only | At least one `.feature` file per API with sunny day + main error cases |
| 7 | Test Cases (enhanced) | Rainy day scenarios, edge cases, and error handling coverage | `code/Test_definitions/` | No | Additional scenarios beyond basic: rainy day, edge cases, error handling |

> | 8 | API description | Wiki page with content for external visibility | CAMARA Sub Project Wiki | No | Wiki page is present and up to date |

Reviewer: ```suggestion
| 8 | API Description | Wiki page with content for external visibility | CAMARA Sub Project Wiki | No | Wiki page is present and up to date |
```

## Requirements by API Status

The following matrix defines which assets are mandatory (M) or optional (O) for each target API status:

| Nr | Asset | alpha | rc | initial public | stable public |
|----|-------|:-----:|:--:|:--------------:|:------------:|
| 1 | Release Plan | M | M | M | M |
| 2 | API Definition(s) | M | M | M | M |
| 3 | Commonalities compliance | O | M | M | M |
| 4 | API Documentation | M | M | M | M |
| 5 | User Stories | O | O | O | M |
| 6 | Test Cases (basic) | O | M | M | M |
| 7 | Test Cases (enhanced) | O | O | O | M |

> | 8 | API description | O | O | M | M |

Reviewer: ```suggestion
| 8 | API Description | O | O | M | M |
```
